pkg/api/v1/conversion.go

```diff
@@ -380,31 +380,7 @@ func Convert_v1_ReplicationControllerSpec_To_api_ReplicationControllerSpec(in *R
 }
 
 func Convert_api_PodStatusResult_To_v1_PodStatusResult(in *api.PodStatusResult, out *PodStatusResult, s conversion.Scope) error {
-	if err := autoConvert_api_PodStatusResult_To_v1_PodStatusResult(in, out, s); err != nil {
-		return err
-	}
-
-	if old := out.Annotations; old != nil {
```

Comment:
Code that uses init containers (which is mostly using versioned clients now) still needs to respect the beta annotation.  So this code needs to move into a function that you can use to, if the annotation is set, override the value that is provided with the object.  For backwards compatibility with APIs "beta" annotations take precedence over fields.